mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1035,6 +1036,50 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			})
 		}
 	})
+
+	// These tests only run when a KMS mock server is running on localhost:8000.
+	mt.RunOpts("kms tls tests", noClientOpts, func(mt *mtest.T) {
+		testcases := []struct {
+			name       string
+			envValue   string
+			errMessage string
+		}{
+			{
+				"invalid certificate",
+				"INVALID_CERT",
+				"expired",
+			},
+			{
+				"invalid hostname",
+				"INVALID_HOSTNAME",
+				"SANs",
+			},
+		}
+
+		for _, tc := range testcases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				// Only run test if correct KMS mock server is running.
+				if os.Getenv("KMS_TLS_TESTCASE") == tc.envValue {
+					ceo := options.ClientEncryption().
+						SetKmsProviders(fullKmsProvidersMap).
+						SetKeyVaultNamespace(kvNamespace)
+					cpt := setup(mt, nil, nil, ceo)
+
+					_, err := cpt.clientEnc.CreateDataKey(context.Background(), "aws", options.DataKey().SetMasterKey(
+						bson.D{
+							{"region", "us-east-1"},
+							{"key", "arn:aws:kms:us-east-1:579766882180:key/89fcc2c4-08b0-4bd9-9f25-e30687b580d0"},
+							{"endpoint", "mongodb://127.0.0.1:8000"},
+						},
+					))
+					assert.True(mt, strings.Contains(err.Error(), tc.errMessage),
```

Comment:
You should first assert that the error is non-nil to ensure that the `err.Error()` call does not panic.

Reply:
Added.